runtime/server/migrations.go

```diff
@@ -37,40 +38,126 @@ func (s *Server) MigrateSingle(ctx context.Context, req *api.MigrateSingleReques
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	// Check if object exists and is not a source
-	obj, found := catalog.FindObject(ctx, req.InstanceId, source.Name)
-	if found && obj.Type != drivers.CatalogObjectTypeSource {
-		return nil, status.Errorf(codes.FailedPrecondition, "an object of type '%s' already exists with name '%s'", obj.Type, obj.Name)
-	}
-
 	// Get olap
 	conn, err := s.cache.openAndMigrate(ctx, inst.ID, inst.Driver, inst.DSN)
 	if err != nil {
 		return nil, status.Error(codes.Unknown, err.Error())
 	}
 	olap, _ := conn.OLAPStore()
 
-	// Ingest the source
-	err = olap.Ingest(ctx, source)
-	if err != nil {
-		return nil, status.Error(codes.Unknown, err.Error())
+	// Get existing object with name and check it's a source
```

Comment:
This method is getting massive. We should break this up i feel.

Reply:
Since it's piped through the `CREATE SOURCE` syntax, it will only parse successfully for valid names. [Here](https://github.com/rilldata/rill-developer/blob/1244637a913595898066c0b99bf892ccbea96968/runtime/sql/pure/pure.go#L12)'s the regex it matches by:

```go
	{Name: `Ident`, Pattern: `[a-zA-Z_][a-zA-Z0-9_]*`},
```